crates/common/tedge_utils/src/notify.rs

```diff
@@ -70,142 +61,77 @@ impl NotifyStream {
+                            },
+                            EventKind::Modify(modify_kind) => match modify_kind {
+                                ModifyKind::Data(_)
+                                | ModifyKind::Metadata(_)
+                                | ModifyKind::Name(_)
+                                | ModifyKind::Any
+                                | ModifyKind::Other => Some(FsEvent::Modified),
+                            },
+                            EventKind::Remove(remove_kind) => match remove_kind {
+                                RemoveKind::File | RemoveKind::Any | RemoveKind::Other => {
+                                    Some(FsEvent::FileDeleted)
                                 }
+                                RemoveKind::Folder => Some(FsEvent::DirectoryDeleted),
+                            },
+                            EventKind::Any | EventKind::Other => Some(FsEvent::Modified),
+                        };
+
+                        if let Some(event) = event {
+                            for path in notify_event.paths {
+                                let _ = tx.send((path, event)).await;
                             }
-                            EventKind::Create(CreateKind::Folder) => {
-                                for path in notify_event.paths {
-                                    let _ = tx.send((path, FsEvent::DirectoryCreated)).await;
-                                }
-                            }
-                            EventKind::Remove(RemoveKind::File) => {
-                                for path in notify_event.paths {
-                                    let _ = tx.send((path, FsEvent::FileDeleted)).await;
-                                }
-                            }
-                            EventKind::Remove(RemoveKind::Folder) => {
-                                for path in notify_event.paths {
-                                    let _ = tx.send((path, FsEvent::DirectoryDeleted)).await;
-                                }
-                            }
-                            _other => {}
                         }
                     }
                 })
             },
             Config::default(),
         )?;
-        Ok(Self {
-            watcher,
-            rx,
-            metadata: HashMap::new(),
-        })
-    }
-
-    #[cfg(test)]
-    fn get_metadata(&self) -> &Metadata {
-        &self.metadata
-    }
-    fn get_metadata_as_mut(&mut self) -> &mut Metadata {
-        &mut self.metadata
+        Ok(Self { watcher, rx })
     }
 
-    pub fn add_watcher(
-        &mut self,
-        dir_path: &Path,
-        file: Option<String>,
-        events: &[FsEvent],
-    ) -> Result<(), NotifyStreamError> {
+    /// Will return an error if you try to watch a file/directory which doesn't exist
+    pub fn add_watcher(&mut self, dir_path: &Path) -> Result<(), NotifyStreamError> {
```

Comment:
Removing the list of watch events is a nice simplification. The caller might receive more messages than expected but it far simpler to do the filtering there.